### ISSUE TITLE
 The vscode extension is inconsistent with the default value of the prettier configuration file

### ISSUE THREAD

**Issue by a81n9** [prettier/prettier-vscode#3453]
I found that the default configuration of the prettier vscode extension is inconsistent with the prettier configuration file.

For example, the default configuration of prettier vscode extended `trailingComma` is `es5`, while the default configuration of prettier configuration file is `all`.

![Prettier vscode Extension](https://github.com/prettier/prettier-vscode/assets/53040248/fe3284f1-812f-460a-85be-f77368bef360)

![Prettier options](https://github.com/prettier/prettier-vscode/assets/53040248/d73a8f02-d780-4b4d-acaf-57256b07284f)

**Comment by timrobinson33** [comment]
I had exactly this problem - it's becuase the version of prettier itself bundled inside the vs code extension is an old version (2.8) which has a different default value for the trailing comma parameter

as suggested somewhere in the documentation, if you `npm install` prettier on your machine (either globally or in the project, the vs code extension will use this version and so you will get the latest defaults

**Comment by jedwards1211** [comment]
Yeah the extension can *use* prettier 3 but that doesn't mean it will change the default format settings in the extension.  Try it in a project that doesn't declare its own format settings, you will see

**Comment by jackbravo** [comment]
To avoid this issue, you can create an empty `.prettierrc` file in the root of your project. That way, when VSCode prettier extension runs it detects that you have an empty configuration an respects it.

I saw this because the logs for the prettier extension said:

```
["INFO" - 5:59:46 PM] Extension Name: esbenp.prettier-vscode.
["INFO" - 5:59:46 PM] Extension Version: 11.0.0.
["INFO" - 6:06:27 PM] Using config file at /Users/jackbravo/Documents/meetearnest/aegis-authentication-service/.prettierrc
["INFO" - 6:29:48 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 6:29:48 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
...
```

**Comment by jackbravo** [comment]
This also seems like a duplicate of #3487, which mentions the same workaround I mentioned :-p. And also another one of setting the `prettier.requireConfig` config option.

**Comment by jedwards1211** [comment]
That's not a good workaround: This happened to me contributing to a project that's not mine... didn't realize until CI failed.  It should just work off the bat in any project we're contributing to.  I'm not sure that project is even willing to have an empty .prettierrc in their repo

**Comment by ntotten** [comment]
This issue should be resolved in version 12 of the extension. Version 12 is currently in preview and is published only to the new Prettier organization. You can find that version here: https://marketplace.visualstudio.com/items?itemName=Prettier.prettier-vscode

Once we decide version 12 is stable it will be published to the old extension location as well.

Please test this issue in version 12 and let me know if this is still an issue. If we don't hear back this issue may be closed.

**Comment by A81N9** [comment]
> This issue should be resolved in version 12 of the extension. Version 12 is currently in preview and is published only to the new Prettier organization. You can find that version here: [marketplace.visualstudio.com/items?itemName=Prettier.prettier-vscode](https://marketplace.visualstudio.com/items?itemName=Prettier.prettier-vscode)
> 
> Once we decide version 12 is stable it will be published to the old extension location as well.
> 
> Please test this issue in version 12 and let me know if this is still an issue. If we don't hear back this issue may be closed.

Thanks for your reply, this issue has been resolved in version 12.

Compared to Oxc and Biome, I still prefer Prettier because the Prettier extension supports direct configuration of global formatting rules. When I edit a single file outside of the workspace, I have a good formatting experience. Thank you.
